src/cli.js

```diff
@@ -235,41 +317,58 @@ function cli(inputArgs) {
-                    var checkSummary = checkItem.name + ': ' +
-                                    (checkItem.installed ? 'installed ' : 'not installed ') +
-                                    (checkItem.metadata.version || '');
-                    events.emit('log', checkSummary);
-                    if (!checkItem.installed) {
-                        someChecksFailed = true;
-                        events.emit('warn', checkItem.metadata.reason);
+                var someChecksFailed = Object.keys(platformChecks).map(function(platformName) {
+                    events.emit('log', '\nRequirements check results for ' + platformName + ':');
+                    var platformCheck = platformChecks[platformName];
+                    if (platformCheck instanceof CordovaError) {
+                        events.emit('warn', 'Check failed for ' + platformName + ' due to ' + platformCheck);
+                        return true;
                     }
+
+                    var someChecksFailed = false;
+                    platformCheck.forEach(function(checkItem) {
+                        var checkSummary = checkItem.name + ': ' +
+                            (checkItem.installed ? 'installed ' : 'not installed ') +
+                            (checkItem.metadata.version || '');
+                        events.emit('log', checkSummary);
+                        if (!checkItem.installed) {
+                            someChecksFailed = true;
+                            events.emit('warn', checkItem.metadata.reason);
+                        }
+                    });
+
+                    return someChecksFailed;
+                }).some(function(isCheckFailedForPlatform) {
+                    return isCheckFailedForPlatform;
                 });
 
-                return someChecksFailed;
-            }).some(function (isCheckFailedForPlatform) {
-                return isCheckFailedForPlatform;
+                if (someChecksFailed) throw new CordovaError('Some of requirements check failed');
             });
-
-            if (someChecksFailed) throw new CordovaError('Some of requirements check failed');
-        }).done();
+        return result.finally(function() {
+            if (shouldCollectTelemetry) {
+                telemetry.track(cmd, result.isFulfilled() ? 'successful' : 'unsuccessful');
+            }
+            return result;
+        });   
     } else if (cmd == 'serve') {
         var port = undashed[1];
-        cordova.raw.serve(port).done();
+        var result = cordova.raw.serve(port);
+        return result.finally(function() {
+            if(shouldCollectTelemetry) {
+                telemetry.track(cmd, result.isFulfilled() ? 'successful' : 'unsuccessful');
+            }
+        });
     } else if (cmd == 'create') {
-        create();
+        var result = create();
+        return result.finally(function() {
+            if (shouldCollectTelemetry) {
```

Comment:
As discussed offline, I'm going ahead with this refactoring. but it will require retesting the whole CLI as it affects everything. Looking at things from the benefits/costs perspective, I'm not sure if that's the best route, but ...